src/coreclr/src/jit/importer.cpp

```diff
@@ -5947,21 +5947,21 @@ int Compiler::impBoxPatternMatch(CORINFO_RESOLVED_TOKEN* pResolvedToken, const B
                     case CEE_UNBOX_ANY:
                         if ((nextCodeAddr + 1 + sizeof(mdToken)) <= codeEndp)
                         {
+                            // See if the resolved tokens in box, isinst and unbox.any describe types that are equal.
                             CORINFO_RESOLVED_TOKEN isinstResolvedToken = {};
                             impResolveToken(codeAddr + 1, &isinstResolvedToken, CORINFO_TOKENKIND_Class);
 
-                            CORINFO_RESOLVED_TOKEN unboxResolvedToken = {};
-                            impResolveToken(nextCodeAddr + 1, &unboxResolvedToken, CORINFO_TOKENKIND_Class);
-
-                            // See if the resolved tokens describe types that are equal.
                             const TypeCompareState compareBoxIsInst =
                                 info.compCompHnd->compareTypesForEquality(isinstResolvedToken.hClass,
                                                                           pResolvedToken->hClass);
+                            CORINFO_RESOLVED_TOKEN unboxResolvedToken = {};
```

Comment:
This doesn't defer the cost of the second set of calls -- resolution and type compare for the unbox token should happen only if the isinst comparsion is successful.